astropy/nddata/nduncertainty.py

```diff
@@ -485,6 +538,206 @@ class _VariancePropagationMixin:
+            # to `_propagate_add_sub`, for example:
+            else:
+                # lookup the mapping for to_variance and from_variance for this
+                # numpy operation:
+                to_variance = collapse_to_variance_mapping[numpy_op]
+                from_variance = collapse_from_variance_mapping[numpy_op]
+
+                if (
+                    self.unit is not None
+                    and to_variance(self.unit) != self.parent_nddata.unit**2
+                ):
+                    # If the uncertainty has a different unit than the result we
+                    # need to convert it to the results unit.
+                    this = to_variance(self.array << self.unit).to(result_unit_sq).value
+                else:
+                    this = to_variance(self.array)
+
+                return from_variance(this, axis=axis)
+
+    @staticmethod
+    def _build_signature(a, axis):
+        """
+        Builds the call signature for np.vectorize in _get_err_at_min or _get_err_at_max.
+        """
+        if a.ndim > 3:
+            raise NotImplementedError(
+                "extrema collapse functions only implemented up to ndim=3."
+            )
+        if not hasattr(axis, "__len__"):
+            # make `axis` iterable
+            axis = [axis]
+        letters = list("abc")
+        input_dimensions = ",".join([letter for letter in letters[: a.ndim]])
+        output_dimensions = ",".join(
+            [letter for i, letter in enumerate(letters) if i not in axis]
+        )
+        if len(output_dimensions) == 0:
+            # this happens when completely collapsing the cube:
+            output_dimensions = 1
+        signature = f"({input_dimensions}),({input_dimensions})->({output_dimensions})"
+        return signature
+
+    @staticmethod
+    def _get_err_at_extremum(measurement, uncertainty, extremum):
+        """
+        Return the value of the ``uncertainty`` array at the indices
+        which satisfy the ``extremum`` function applied to the ``measurement`` array,
+        where we expect ``extremum`` to be np.argmax or np.argmin, and
+        we expect a two-dimensional output.
+
+        Assumes the ``measurement`` and ``uncertainty`` array dimensions
+        are ordered such that the zeroth dimension is the one to preserve.
+        For example, if you start with array with shape (a, b, c), this
+        function applies the ``extremum`` function to the last two dimensions,
+        with shapes b and c.
+
+        This operation is difficult to cast in a vectorized way. Here
+        we implement it with a list comprehension, which is likely not the
+        most performant solution.
+        """
+        return np.array(
+            [
+                err[np.unravel_index(extremum(meas), meas.shape)]
+                for meas, err in zip(measurement, uncertainty)
+            ]
+        )
+
+    def _get_err_at_min(self, axis):
+        supported_axes = [0, 1, 2]
```

Comment:
Does this mean that an NDData object can only be collapsed over its first 3 dimensions?

Reply:
This is true in the current implementation, yes.

Reply:
OK.  Thanks for clarifying.  Unfortunately this would cause problems when applying this multiple several solar use cases where we often have more than 3 dimensions.  Perhaps the most high profile would be DKIST data analysis.